heron/metricsmgr/tests/java/com/twitter/heron/metricsmgr/sink/metricscache/MetricsCacheSinkTest.java

```diff
@@ -118,9 +120,9 @@ public void testMetricsCacheClientService() throws Exception {
 
     // Then we check whether the MetricsCacheService has restarted the MetricsCacheClient for
     // several times Take other factors into account, we would check whether the MetricsCacheClient
-    // has restarted at least half the RESTART_WAIT_INTERVAL_SECONDS/RECONNECT_INTERVAL
+    // at least RESTART_WAIT_INTERVAL - RECONNECT_INTERVAL * 2
     assertTrue(metricsCacheSink.getMetricsCacheStartedAttempts()
-        >= (RESTART_WAIT_INTERVAL.getSeconds() / RECONNECT_INTERVAL.getSeconds() / 2));
+        >= (RESTART_WAIT_INTERVAL.getSeconds() - 2 * RECONNECT_INTERVAL.getSeconds()));
```

Comment:
Isn't `metricsCacheStartedAttempts` the number of attempts?

Reply:
Yeah. It is the number of attempts to connect to a specific endpoint.

Reply:
In fact, let me think about this line again. The idea for the old and new code are similar and the new code focus on the first time. I may consider reverting this line. Thanks for the question.

Reply:
Yeah, I was feeling it's wired to compare (x times >= y seconds).